library/std/src/sys/pal/windows/futex.rs

```diff
@@ -0,0 +1,52 @@
+use super::api;
+use crate::sys::c;
+use crate::sys::dur2timeout;
+use core::ffi::c_void;
+use core::mem;
+use core::ptr;
+use core::time::Duration;
+
+#[inline(always)]
```

Comment:
This is not a hot path, and a private-only API, so I think it's better to leave this up to the optimizer:
```suggestion
```